src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -305,43 +321,219 @@ function compute_gwd_fields(domain) result(iErr)
          end if
       end do
 
-
-      deallocate(topo)
-      deallocate(landuse)
       deallocate(hlanduse)
 
-      iErr = 0
+      iErr = free_tile_list(tilesHead)
 
    end function compute_gwd_fields
 
 
    !***********************************************************************
    !
-   !  function read_global_30s_topo
+   !  subroutine get_box_size_from_lat_lon
+   !
+   !> \brief   Routine to obtain box size given the mean diameter (meters), lat, lon (radians)
+   !> \author  Abishek Gopal
+   !> \date    05 Sep 2024
+   !> \details 
+   !>  Routine to obtain box size (nx, ny) given the mean diameter of the grid cell (meters),
+   !     and the latitude and longitude coordinates (radians)
    !
-   !> \brief   Reads global 30-arc-second topography into 'topo' module variable
+   !-----------------------------------------------------------------------
+   subroutine get_box_size_from_lat_lon(lat, lon, dx, nx, ny)
+      
+      implicit none
+
+      real (kind=RKIND), intent(in) :: lat
+      real (kind=RKIND), intent(in) :: lon
+      real (kind=RKIND), intent(in) :: dx
+      integer, intent(out) :: nx
+      integer, intent(out) :: ny
+
+      !
+      ! Get number of points to extract in the zonal direction
+      !
+      if (cos(lat/rad2deg) > (2.0 * pts_per_degree * dx * 180.0) / (real(topo_x,RKIND) * Pi * Re)) then
+         nx = ceiling((180.0 * dx * pts_per_degree) / (Pi * Re * cos(lat/rad2deg)))
+      else
+         nx = topo_x / 2
+      end if
+
+      !
+      ! Get number of points to extract in the meridional direction
+      !
+      ny = ceiling((180.0 * dx * pts_per_degree) / (Pi * Re))
+
+
+   end subroutine get_box_size_from_lat_lon
+
+
+
+   !***********************************************************************
+   !
+   !  function get_tile_from_box_point
+   !
+   !> \brief   Routine to obtain a tile given a box pixel
+   !> \author  Abishek Gopal
+   !> \date    05 Sep 2024
+   !> \details 
+   !>  Routine to obtain a tile of type mpas_gwd_tile_type, given the linked 
+   !    list of tiles tilesHead, box coordinates box_x, box_y, and path to 
+   !    static dataset. The function first searches the linked list to locate
+   !    the tile, and if search fails, adds the target tile to the linked list
+   !    after reading in the data for the tile from disk 
+   !-----------------------------------------------------------------------
+   function get_tile_from_box_point(tilesHead, box_x, box_y, path, sub_path) result(thisTile)
+
+      implicit none
+
+      integer, intent(in) :: box_x, box_y
+      type(mpas_gwd_tile_type), pointer, intent(in) :: tilesHead
+      type(mpas_gwd_tile_type), pointer :: thisTile
+      character(len=*), intent(in) :: path
+      character(len=*), intent(in) :: sub_path
```

Comment:
I think it's preferable to declare in this order:
1. Dummy arguments
2. Return value
3. Local variables